### DISCUSSION TITLE
Table extraction

### DISCUSSION THREAD

**Discussion by erkin98** [jsvine/pdfplumber#406] [Q&A] [answered]
i want to extract this table(in red circle) in this [photo](https://drive.google.com/file/d/1ap3PQ836IUSeSilXtd9YXflKbQBrLYQC/view?usp=sharing) as a separate table
which parameters i should use?

**Comment by samkit-jain** [comment]
I think the default setting of `lines` should work fine. Could you please share the PDF as well?

**Comment by erkin98** [comment]
[Here you go](https://drive.google.com/file/d/1HRSEz7VJbTctY4vBehohLF_446BMxrKa/view?usp=sharing)

**Reply by erkin98** [reply]
lines works fine but it returns merged with middle table but i want to them separate

**Comment by samkit-jain** [comment, marked as answer]
Thanks for sharing the PDF. The tables are getting merged in because they are contiguous. One option could be to remove the `None` values from the rows and then separate them as 2 tables like
```python
for row in merged_table:
    non_null_row = [cell for cell in row if cell is not None]
    
    if len(non_null_row) > 2:
        middle_table.append(non_null_row)
    else:
        bottom_table.append(non_null_row)
```

Does it solve your use case?

**Reply by erkin98** [reply]
absolutely solve my case thanks for helping
